api/kyverno/v1/common_types.go

```diff
@@ -172,6 +172,9 @@ type ServiceCall struct {
 	// the server certificate.
 	// +kubebuilder:validation:Optional
 	CABundle string `json:"caBundle" yaml:"caBundle"`
+
+	// Secret is a Reference to a Secret resource that contains a CABundle
+	Secret *SecretReference `json:"secret,omitempty" yaml:"secret,omitempty"`
```

Comment:
`secret` is too vague, you should pick up a better name IMHO.

Reply:
If a CaBundle and secret is provided then we use the caBundle. 
IMO, hardcoded caBundle should take priority

Reply:
How about adding a validation rule to make sure only one is set ?

Reply:
Yep, that the better thing to do, I am adding it

Reply:
I have renamed the field from `secret` to `tlssecret` as these secrets are of type `kubernetes.io/tls`

Reply:
to me `caBundleFromSecret` sounds a better name 🤷 
ideally `caBundle` could be a struct but that's probably not an option because it would introduce a breaking change.